node.go

```diff
@@ -358,14 +385,19 @@ func (n *ProtoNode) Multihash() mh.Multihash {
 	return n.cached.Hash()
 }
 
-// Links returns the node links.
+// Links returns a copy of the node's links.
 func (n *ProtoNode) Links() []*format.Link {
-	return n.links
+	links := make([]*format.Link, len(n.links))
+	copy(links, n.links)
+	return links
 }
```

Comment:
This seems more reasonable. Hopefully no one is using mutation here, did you check for usage here or mostly assuming folks haven't been doing anything nuts here?

Reply:
No, I haven't checked very broadly and I haven't seen anyone doing anything weird but I don't want to expose this if we're taking so much care with statefulness of sorting internally. But, I had to do this here https://github.com/filecoin-project/boost/pull/675 to deal with the inverse - that go-merkledag was messing with it.